### PR TITLE
Bulk download invoices

### PR DESCRIPTION
**Pull request by Shruti-Apte**
## Notion card
https://www.notion.so/saeloun/Frontend-Bulk-Invoice-download-dc97201519554ec8a9e4698a7b3cb865
## Summary

- Bulk download functionality UI and Integration.
## Preview

https://www.loom.com/share/375049adc7134d32b30206eb1859fb49
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- Checked bulk download functionality.
- Need to test all other workflows like single download.
- -Backend Pr is not yet merged for this nor reviewd.
### Checklist:

- [ ] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@Shruti-Apte Couple of fixes we need to make. Please seem the loom video https://www.loom.com/share/171f20c67b9d411e9813f618d1e905bf cc @supriya3105

File: app/services/bulk_invoice_download_service.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  def process
+    zip_invoices
+  end
+
+  private
+
+    def zip_invoices
+      zipper = Zipper.new(invoices_temp_pdf_data)
+      zipper.zip
+      file_uploader = FileUploader.new(zipper.file, "Invoices_" + Time.now.to_i.to_s + ".zip")
```
[the remark this answers is not in the document]
Comment (shalapatil) [reply app/services/bulk_invoice_download_service.rb]: done

File: lib/file_uploader.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class FileUploader
```
[the remark this answers is not in the document]
Comment (shalapatil) [reply lib/file_uploader.rb]: done

**Review by apoorv1316 — APPROVED** [review]

**Review by prasanthchaduvula — APPROVED** [review]

File: spec/lib/file_uploader_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  describe "#delete_after" do
+    before do
+      @file_uploader = FileUploader.new(Tempfile.new(), "abc")
+      @blob_obj = @file_uploader.upload
+      @file_uploader.delete_after(1.seconds)
+    end
+
+    it "check if DeleteActiveStorageBlobJob job is queued" do
```
[the remark this answers is not in the document]
Comment (shalapatil) [reply spec/lib/file_uploader_spec.rb:44]: will do this in separate PR

**Pull request merged**